integrationtest/mapper/orm/src/test/java/org/hibernate/search/integrationtest/mapper/orm/massindexing/MassIndexingEmbeddedIdIT.java

```diff
@@ -0,0 +1,234 @@
+ */
+public class MassIndexingEmbeddedIdIT {
+
+	public static final String TITLE_1 = "Oliver Twist";
+	public static final String AUTHOR_1 = "Charles Dickens";
+	public static final String TITLE_2 = "Ulysses";
+	public static final String AUTHOR_2 = "James Joyce";
+	public static final String TITLE_3 = "Frankenstein";
+	public static final String AUTHOR_3 = "Mary Shelley";
+
+	@Rule
+	public BackendMock backendMock = new BackendMock( "stubBackend" );
+
+	@Rule
+	public OrmSetupHelper ormSetupHelper = new OrmSetupHelper();
+
+	private SessionFactory sessionFactory;
+	private Book book1;
+	private Book book2;
+	private Book book3;
+
+	@Before
+	public void setup() {
+		backendMock.expectAnySchema( Book.INDEX );
+
+		sessionFactory = ormSetupHelper.withBackendMock( backendMock )
+				.withPropertyRadical( HibernateOrmMapperSettings.Radicals.AUTOMATIC_INDEXING_STRATEGY, HibernateOrmAutomaticIndexingStrategyName.NONE )
+				.setup( Book.class );
+
+		backendMock.verifyExpectationsMet();
+
+		initData();
+	}
+
+	@Test
+	public void defaultMassIndexerStartAndWait() {
+		OrmUtils.withinSession( sessionFactory, session -> {
+			SearchSession searchSession = Search.getSearchSession( session );
+			MassIndexer indexer = searchSession.massIndexer();
+
+			// add operations on indexes can follow any random order,
+			// since they are executed by different threads
+			backendMock.expectWorksAnyOrder(
+					Book.INDEX, DocumentCommitStrategy.NONE, DocumentRefreshStrategy.NONE
+			)
+					.add( book1.getId().toString(), b -> b
+							.field( "title", TITLE_1 )
+							.field( "author", AUTHOR_1 )
+					)
+					.add( book2.getId().toString(), b -> b
+							.field( "title", TITLE_2 )
+							.field( "author", AUTHOR_2 )
+					)
+					.add( book3.getId().toString(), b -> b
+							.field( "title", TITLE_3 )
+							.field( "author", AUTHOR_3 )
+					)
+					.preparedThenExecuted();
+
+			// purgeAtStart, optimizeAfterPurge and purgeAtStart flags are active by default,
+			// so we expect 1 purge, 2 optimize and 1 flush calls in this order:
+			backendMock.expectIndexScopeWorks( Book.INDEX, session.getTenantIdentifier() )
+					.purge()
+					.optimize()
+					.optimize()
+					.flush();
+
+			try {
+				indexer.startAndWait();
+			}
+			catch (InterruptedException e) {
+				fail( "Unexpected InterruptedException: " + e.getMessage() );
+			}
+
+		} );
+
+		backendMock.verifyExpectationsMet();
+	}
+
+	private void initData() {
+		OrmUtils.withinTransaction( sessionFactory, session -> {
+			book1 = new Book( 1, TITLE_1, AUTHOR_1 );
+			session.persist( book1 );
+			book2 = new Book( 2, TITLE_2, AUTHOR_2 );
+			session.persist( book2 );
+			book3 = new Book( 3, TITLE_3, AUTHOR_3 );
+			session.persist( book3 );
+		} );
+	}
+
+	@Entity
+	@Table(name = "book")
+	@Indexed(index = Book.INDEX)
+	public static class Book {
+		public static final String INDEX = "Book";
+
+		@EmbeddedId
+		private BookId id;
```

Comment:
Could you maybe give another name to the ID property, such as "foobar"? I'm not sure using "id" in a query string will have Hibernate ORM auto-magically unrestand we're refering to the ID.

In particular, what if another property is named "id", but is not the entity id?

Reply:
Good idea!

Reply:
You're right! If I change the name of the Hibernate ORM identifier, it works. But if I add a non-identifier field having `id` as a name, it doesn't!